Bonfire-Caesars-Cipher.md

```diff
@@ -48,6 +48,41 @@ function rot13(str) {
 }
 ```
 
+#### Second solution (advanced)
+
+```js
+function rot13(str) {
+    
+    var newStr="";
+    
+    var map = Array.prototype.map;
+  
+    // Make an array with ASCII values from str
+    var strArray = [].map.call(str, function(x) {
+        return x.charCodeAt(0); 
+    });
+  
+    // Take the previous array (strArray) and then shift all carracters codes 
+    // and place them in a string variable
+    map.call(strArray, function(x) {
+        //Checks if character lies between A-Z
+        if (x < 65 || x > 91) {
+            newStr += String.fromCharCode(x);
```

Comment:
This is a bad an incorrect use of map.

If you're modifying data outside of the map, you should be using `.forEach`.